solutions/hw5/treap.py

```diff
@@ -0,0 +1,122 @@
+from collections import MutableMapping
+from typing import Iterator, Callable, Any
+from random import randint
+
+
+class Node:
+    priority: int
+
+    def __init__(self, key, value):
+        self.key = key
+        self.value = value
+        self.left_child = None
+        self.right_child = None
+        self.priority = randint(0, 100)
+
+    def inorder(self) -> list:
+        values = []
+        if self.have_left_child():
+            values.extend(self.left_child.inorder())
+        values.append(self.value)
+        if self.have_right_child():
+            values.extend(self.right_child.inorder())
+        return values
+
+    def have_left_child(self):
+        return self.left_child is not None
+
+    def have_right_child(self):
+        return self.right_child is not None
+
+
+def split(node: Node, key):
+    if node is None:
+        return None, None
+    elif key > node.key:
+        left, right = split(node.right_child, key)
+        node.right_child = left
+        return node, right
+    elif key < node.key:
+        left, right = split(node.left_child, key)
+        node.left_child = right
+        return left, node
+    else:
+        return node.left_child, node.right_child
+
+
+def merge(small_keys_s_tree: Node, big_keys_s_tree: Node):
+    if small_keys_s_tree is None:
+        return big_keys_s_tree
+    if big_keys_s_tree is None:
+        return small_keys_s_tree
+    if small_keys_s_tree.priority > big_keys_s_tree.priority:
+        small_keys_s_tree.right_child = merge(small_keys_s_tree.right_child, big_keys_s_tree)
+        return small_keys_s_tree
+    else:
+        big_keys_s_tree.left_child = merge(small_keys_s_tree, big_keys_s_tree.left_child)
+        return big_keys_s_tree
+
+
+class Deramida(MutableMapping):
+    root: Node
+    __size: int
+
+    def __init__(self):
+        self.__size = 0
+        self.root = None
+
+    def __contains__(self, key):
+        try:
+            a = self[key]
+            return True
+        except KeyError:
+            return False
```

Comment:
It looks a little bit strange. Why do you use unused variable a? It is a little bit strange approach to use try...catch block to check if the key exists